### DISCUSSION TITLE
Tap on left/right for page up/down when reading article (alternative to scrolling)

### DISCUSSION THREAD

**Discussion by lukeuser** [jocmp/capyreader#497] [Feature Requests]
I use capyreader on my eink device. The feature I'm suggesting is particularly suited to eink devices, since scrolling on an eink display is never ideal.

**I would love the option that, instead of scrolling, progressing through the article is done by tapping somewhere on the right of the screen (this could be the whole 25% of the viewing area). Tapping would then jump the content down by most of the height of the viewing area. The opposite would happen when tapping on the left side.**

Scrolling wouldn't need to be disabled; it might still be wanted to centre a specific part of the content (though it perhaps could be limited to the central area?).

The idea here is to emulate the experience of reading an ebook, tapping to progress to the next page. One app that already does this (with web content) is the e-ink web browser [einkbro](https://github.com/plateaukao/einkbro).

Many thanks for considering this. I really love this app, and it (despite not having this feature) works very well on my eink device.

**Comment by jocmp** [comment]
hey @lukeuser - thanks for sharing. i do want to improve the app for e-ink, especially right-to-left nav like you mentioned. I'll take a look at the einkbro library. It looks promising!

**Reply by jocmp** [reply]
Just a prototype right now but here's an approximation of Einkbro's functionality. Let me know what you think

- Tap bottom right to move forward by a page
- Tap bottom left to move back by a page
- Long press left or right to move to document top or bottom respectively

https://github.com/user-attachments/assets/b3fce07c-2e0b-4bde-a11d-4db7636d1fda

**Reply by lukeuser** [reply]
Thanks so much for taking this up!!

 In terms of the prototype, that's perfect—exactly what I had in mind.

The long-press options are very nice to have too.

The only thing to add is regarding the size (or location) of the areas to tap. Holding an eink tablet-type device, you wouldn't want to have to reach down too much to the bottom to tap. If the areas you're using are similar to the default ones used on einkbro, that's totally fine. The whole left/right sides was my instinct, but I can see why not having the areas take up the whole height makes sense (if you want to follow a link that's near the side, for example).

**Reply by jocmp** [reply]
I definitely agree with having the tap areas only take up part of the screen. They're invisible for Capy - similar to an eBook reader - but I've colored them in below for reference. Magenta is the back gesture and cyan is forward. I tried to make the proportions of each area similar to einkbro.

| einkbro | capy prototype |
|--|--|
| ![Screenshot_20241113-162634](https://github.com/user-attachments/assets/ceca405b-0d17-4357-82cf-85a7ba474498) | ![Screenshot_20241113_163808](https://github.com/user-attachments/assets/7203a715-bb02-42da-bf2a-509f13e0b188) |

**Reply by lukeuser** [reply]
I don't see any problem those those areas; looks good!

**Reply by jocmp** [reply]
This is live in the latest version [2024.11.1074-dev](https://github.com/jocmp/capyreader/releases/tag/2024.11.1074-dev). If you use Google Play or the main F-Droid repo it should hit those sources some time next week. Thanks again for the suggestion!
